azurelinuxagent/common/protocol/util.py

```diff
@@ -34,36 +32,25 @@
 from azurelinuxagent.common.future import ustr
 from azurelinuxagent.common.osutil import get_osutil
 from azurelinuxagent.common.dhcp import get_dhcp_handler
+from azurelinuxagent.common.protocol.metadata_server_migration_util import cleanup_metadata_server_artifacts, \
+                                                                           is_metadata_server_artifact_present
 from azurelinuxagent.common.protocol.ovfenv import OvfEnv
 from azurelinuxagent.common.protocol.wire import WireProtocol
-from azurelinuxagent.common.protocol.metadata import MetadataProtocol
 from azurelinuxagent.common.utils.restutil import KNOWN_WIRESERVER_IP, \
                                                   IOErrorCounter
 
 OVF_FILE_NAME = "ovf-env.xml"
-TAG_FILE_NAME = "useMetadataEndpoint.tag"
 PROTOCOL_FILE_NAME = "Protocol"
 MAX_RETRY = 360
```

Comment:
Are we able to reduce MAX_RETRY? In the case that we update from MetadataServer to WireServer, and WireServer is not reachable, we will keep trying to contact it 360 times (1 hour if we probe every 10 seconds) before failing update and reverting to MetadataServer. Previously the retry loop would also try contacting MetadataServer, but since this is stripped out we are now only trying to contact WireServer in _detect.

Reply:
If you want to change the number of retries only during the migration then it should be fine.

Changing it for all other scenarios may not be right, we would need to evaluate whether that would affect public. For that I would recommend doing it on a separate PR.

Reply:
In_detect_protocol in util.py, can we do something this?
retries =  5 if is_metadata_server_artifact_present() else MAX_RETRY